### PR TITLE
add workflow status field to API

### PR DESCRIPTION
**Pull request by chamilaadhi**
- add an extra column to AM_API schema to persist api lifecycle state change workflow status
- add workflow status info to the rest api response object

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/models/API.java [hunk 1]
```diff
@@ -418,6 +427,7 @@ public BusinessInformation getBusinessInformation() {
         private Map<String, UriTemplate> uriTemplates = Collections.EMPTY_MAP;
         private String copiedFromApiId;
         private String apiDefinition;
+        private String workflowStatus;
         public APIBuilder(String provider, String name, String version) {
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/models/API.java]: better to have  space between variable declaration and the constructor

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/models/API.java [hunk 2]
```diff
@@ -289,6 +291,13 @@ public int hashCode() {
     private final String apiDefinition;
     private final HashMap permissionMap;
     private final String apiPermission;
+    private final String workflowStatus;
+    
+
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/models/API.java]: too much space lines

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/models/Workflow.java [hunk 3]
```diff
@@ -135,11 +137,20 @@ public void setAttributes(Map<String, String> attributes) {
...
+        this.createdBy = createdBy;
+    }
+
     @Override
     public String toString() {
         return "Workflow [workflowReference=" + workflowReference + ", workflowType=" + workflowType + ", status="
                 + status + ", createdTime=" + createdTime + ", updatedTime=" + updatedTime + ", workflowDescription="
-                + workflowDescription + ", callbackUrl=" + callbackURL + ", attributes=" + attributes + "]";
-    }    
+                + workflowDescription + ", externalWorkflowReference=" + externalWorkflowReference + ", callbackURL="
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/models/Workflow.java]: Shall we use String builder for this as following.
Example : 
String a = b + c + d;
into
String a = new StringBuilder(b).append(c).append(d).toString();

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/models/API.java (hunk 2 above)
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/models/API.java]: done

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/models/API.java (hunk 1 above)
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/models/API.java]: done

**Pull request merged**